dbms/src/Interpreters/Context.cpp

```diff
@@ -204,8 +206,9 @@ struct ContextShared
 
     Context::ConfigReloadCallback config_reload_callback;
 
-    explicit ContextShared(std::shared_ptr<IRuntimeComponentsFactory> runtime_components_factory_)
+    explicit ContextShared(std::shared_ptr<IRuntimeComponentsFactory> runtime_components_factory_, const std::optional<ServerInfo> & server_info_ = std::nullopt)
```

Comment:
It seems that we don't have to init server_info in constructor.

Reply:
updated